src/qibolab/_core/__init__.py

```diff
@@ -0,0 +1,34 @@
+from . import (
+    backends,
+    components,
+    dummy,
+    execution_parameters,
+    parameters,
+    platform,
+    pulses,
+    qubits,
+    sequence,
+    sweeper,
+)
+from .backends import *
+from .components import *
+from .dummy import *
+from .execution_parameters import *
+from .parameters import *
+from .platform import *
+from .pulses import *
+from .qubits import *
+from .sequence import *
+from .sweeper import *
+
```

Comment:
Shall we include something from `qibolab.identifier` here? `ChannelId` is used in platform creation, for example https://github.com/qiboteam/qibolab_platforms_qrc/blob/9d32819c6f8c1df45549e5cc7a0a4e2b387a5bed/qw11qD/platform.py#L43 (the same for `QubitMap` which is currently defined in `qibolab.platform.platform`). 

In the platform creation these are only used for type annotation, so probably not very important and we could even keep them internal and not use them in the platforms. I doubt users writing platform will care much about type annotation.

However, the main issue is qibocal, where type annotation is somewhat more important and `QubitId` and `QubitPairId` are used all over the place. These are in `qibolab.identifier` and hence currently private. Shall we consider exposing them? I would say that is the simplest thing to do at this stage.

Reply:
In general, these are aliases used mainly internally. Then, they could be used even outside, as they are in Qibocal, and some platforms.

My decision has been to keep them private, for the time being, since exposing them at a later stage is a non-breaking change. While the opposite is breaking.

In any case, if you have a stronger opinion, we could expose them immediately.

Reply:
If we don't expose them, what is a potential replacement for qibocal? @andrea-pasquale told me that they are only used as type hints, so they are not fully required, however I would guess that in many cases they are used as type hints in dataclasses which are quite useful.

I wouldn't call it strong opinion, it just seems that exposing them would make the qibocal update simpler, as we could stick to using them as they are used now.

Reply:
> If we don't expose them, what is a potential replacement for qibocal? @andrea-pasquale told me that they are only used as type hints, so they are not fully required, however I would guess that in many cases they are used as type hints in dataclasses which are quite useful.

Just use `str`, `dict[str, ...]` and so on.
Moreover, most of the time, you don't need type hints for what you're calling, you need to annotate your own objects. Simply, for the time being, the annotations would be "less specific", but not less correct.

Another option is that Qibocal defines its own `QubitId` and so on, possibly duplicating part of the (little) content of the `identifier` module.
Whenever we'll stabilize the types as well, it will very simple to replace ones with the others.

> I wouldn't call it strong opinion, it just seems that exposing them would make the qibocal update simpler, as we could stick to using them as they are used now.

Well, I said *stronger*, not necessarily in absolute terms.
My point is just caution, since I'd like to roll out a small (as possible) public interface, and expand it slowly, but in a reasoned and sustainable way.

Reply:
I would say that readability wise having `QubitId` would be nicer.

Reply:
> Another option is that Qibocal defines its own `QubitId` and so on, possibly duplicating part of the (little) content of the `identifier` module. Whenever we'll stabilize the types as well, it will very simple to replace ones with the others.

If we decide to not expose it from qibolab, I would probably go with that option. In particular, I will define
```py
QubitId = Union[int, str]
QubitPairId = tuple[QubitId, QubitId]
```
somewhere in qibocal and import them wherever they are needed. I mainly need that so that I can test.